ivc/src/ivc/columns.rs

```diff
@@ -284,78 +293,90 @@ impl ColumnIndexer for IVCColumn {
     // This should be
     //   const N_COL: usize = std::cmp::max(IVCPoseidonColumn::N_COL, FECColumn::N_COL);
     // which is runtime-only expression..?
-    const N_COL: usize = IVCPoseidonColumn::N_COL;
+    // 333 is not enough
+    const N_COL: usize = 400;
 
     fn to_column(self) -> Column {
         match self {
+            IVCColumn::BlockSel(i) => {
+                assert!(i < 2 * N_BLOCKS);
+                Column::FixedSelector(i)
+            }
+
             IVCColumn::Block1Input(i) => {
                 assert!(i < 2 * N_LIMBS_SMALL);
-                Column::Relation(i)
+                Column::Relation(N_BLOCKS + i)
             }
             IVCColumn::Block1InputRepacked75(i) => {
                 assert!(i < 2 * N_LIMBS_LARGE);
-                Column::Relation(2 * N_LIMBS_SMALL + i)
+                Column::Relation(N_BLOCKS + 2 * N_LIMBS_SMALL + i)
             }
             IVCColumn::Block1InputRepacked150(i) => {
                 assert!(i < 2 * N_LIMBS_XLARGE);
-                Column::Relation(2 * N_LIMBS_SMALL + 2 * N_LIMBS_LARGE + i)
+                Column::Relation(N_BLOCKS + 2 * N_LIMBS_SMALL + 2 * N_LIMBS_LARGE + i)
             }
 
-            IVCColumn::Block2Hash(poseidon_col) => poseidon_col.to_column(),
+            IVCColumn::Block2Hash(poseidon_col) => {
+                poseidon_col.to_column().add_rel_offset(N_BLOCKS)
+            }
 
-            IVCColumn::Block3ConstPhi => Column::Relation(0),
-            IVCColumn::Block3ConstR => Column::Relation(1),
-            IVCColumn::Block3PhiPow => Column::Relation(2),
-            IVCColumn::Block3PhiPowR => Column::Relation(3),
-            IVCColumn::Block3PhiPowR2 => Column::Relation(4),
-            IVCColumn::Block3PhiPowR3 => Column::Relation(5),
+            IVCColumn::Block3ConstPhi => Column::Relation(N_BLOCKS),
+            IVCColumn::Block3ConstR => Column::Relation(N_BLOCKS + 1),
```

Comment:
To me it looks like `IVCColumn::Block1Input(i)` and `IVCColumn::Block3ConstR` could collide for example when `i=1`, no?

Reply:
It makes sense to collide. It is like selectors in vanilla plonk. The columns are shared.

Reply:
I am not sure I understand why you want redundant definitions of the same column. Like, it is not as if you could access multiple rows for each constraint no? Like, at most the current row and the next, but not 4 rows ahead...

Reply:
I don't understand. We can talk about it in a call, easier.